src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,881 @@
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const StopSaveGraph::Request& req, StopSaveGraph::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const HardForkInfo::Request& req, HardForkInfo::Response& res)
+  {
+    const Blockchain &blockchain = m_core.get_blockchain_storage();
+    uint8_t version = req.version > 0 ? req.version : blockchain.get_ideal_hard_fork_version();
+    res.info.version = blockchain.get_current_hard_fork_version();
+    res.info.enabled = blockchain.get_hard_fork_voting_info(version, res.info.window, res.info.votes, res.info.threshold, res.info.earliest_height, res.info.voting);
+    res.info.state = blockchain.get_hard_fork_state();
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetBans::Request& req, GetBans::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const SetBans::Request& req, SetBans::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const FlushTransactionPool::Request& req, FlushTransactionPool::Response& res)
+  {
+    res.status = Message::STATUS_FAILED;
+    res.error_details = "RPC method not yet implemented.";
+  }
+
+  void DaemonHandler::handle(const GetOutputHistogram::Request& req, GetOutputHistogram::Response& res)
+  {
+    std::map<uint64_t, std::tuple<uint64_t, uint64_t, uint64_t> > histogram;
+    try
+    {
+      histogram = m_core.get_blockchain_storage().get_output_histogram(req.amounts, req.unlocked, req.recent_cutoff);
+    }
+    catch (const std::exception &e)
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = e.what();
+      return;
+    }
+
+    res.histogram.clear();
+    res.histogram.reserve(histogram.size());
+    for (const auto &i: histogram)
+    {
+      if (std::get<0>(i.second) >= req.min_count && (std::get<0>(i.second) <= req.max_count || req.max_count == 0))
+        res.histogram.emplace_back(output_amount_count{i.first, std::get<0>(i.second), std::get<1>(i.second), std::get<2>(i.second)});
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetOutputKeys::Request& req, GetOutputKeys::Response& res)
+  {
+    for (const auto& i : req.outputs)
+    {
+      crypto::public_key key;
+      rct::key mask;
+      bool unlocked;
+      m_core.get_blockchain_storage().get_output_key_mask_unlocked(i.amount, i.index, key, mask, unlocked);
+      res.keys.emplace_back(output_key_mask_unlocked{key, mask, unlocked});
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(const GetRPCVersion::Request& req, GetRPCVersion::Response& res)
+  {
+    res.version = DAEMON_RPC_VERSION;
```

Comment:
This should be renamed wiht 0MQ in the name or it'll get confused with the existing one (as I just did).